linkerd/app/core/src/config.rs

```diff
@@ -46,3 +63,21 @@ impl Param<Keepalive> for ServerConfig {
         self.keepalive
     }
 }
+
+// === impl PortHasher ===
+
+impl Hasher for PortHasher {
+    fn write(&mut self, _: &[u8]) {
+        unreachable!("hashing a `u16` calls `write_u16`");
+    }
+
+    #[inline]
+    fn write_u16(&mut self, port: u16) {
+        self.0 = port as u64;
+    }
```

Comment:
Any reason `PortHasher` has a `u64` meaning we have to cast for every write, instead of it having a `u16` and only casting to `u64` in `finish`?

Reply:
an individual `PortHasher` will only ever see one `write_u16` call; it should be equivalent either way? but can do.